api/v1alpha1/controlplane_types.go

```diff
@@ -22,10 +22,18 @@ import (
 
 // ControlPlaneSpec defines the desired state of ControlPlane
 type ControlPlaneSpec struct {
-	Type               ControlPlaneType  `json:"type,omitempty"`
-	Backend            BackendDBType     `json:"backend,omitempty"`
-	PostCreateHook     *string           `json:"postCreateHook,omitempty"`
-	PostCreateHookVars map[string]string `json:"postCreateHookVars,omitempty"`
+	Type    ControlPlaneType `json:"type,omitempty"`
+	Backend BackendDBType    `json:"backend,omitempty"`
+	// BootstrapSecretRef contains a reference to the kubeconfig used to bootstrap adoption of
+	// an external cluster
+	// +optional
+	BootstrapSecretRef *SecretReference `json:"bootstrapSecretRef,omitempty"`
+	// expiration time for generated auth token
```

Comment:
A comment on a public field of a public type should begin with the name of the field. This is a general rule of the Go language. FYI, in the Kubernetes world there is a preference --- in the case of part of an API object --- to cite the name as it appears on JSON (initial lowercase) rather than as it appears in the Go source (initial uppercase) and code-quoted with backticks; but this is not strictly followed.

Reply:
updated in latest commit.